rust-runtime/aws-smithy-wasm/src/wasm.rs

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! WASM Tools for the SDK
+use aws_smithy_async::rt::sleep::{AsyncSleep, Sleep};
+
+/// A struct implementing the AsyncSleep trait that can be used in
+/// Wasm environments
+#[derive(Debug, Clone)]
+pub struct WasmSleep;
+
+impl AsyncSleep for WasmSleep {
+    fn sleep(&self, duration: std::time::Duration) -> Sleep {
+        Sleep::new(Box::pin(async move {
+            tokio::time::sleep(duration).await;
```

Comment:
Why have a separate `WasmSleep` if this is essentially identical to the `TokioSleep` in aws-smithy-async? Shouldn't customers also have the option to use a different async runtime in WASM?

Reply:
Had no idea `TokioSleep` existed, will remove this in favor of that!

Reply:
Addressed in https://github.com/smithy-lang/smithy-rs/pull/3409/commits/c883aad007e0f5929794be8f5370854679a5f213